WordPress/gutenberg
# Convert FocalPointPicker tests to TypeScript

## What?
This PR converts the `index` and `media` tests for the `FocalPointPicker` to TypeScript.

## Why?
Adding TypeScript support resolves this issue: https://github.com/WordPress/gutenberg/issues/60529

## How?
- converted both files to `.tsx`
- fixed missing property errors by passing default `{ ...props }` to components (similar to [this test](https://github.com/WordPress/gutenberg/blob/c56672504bada434aee1ec331366cd2a10ab2b38/packages/components/src/border-box-control/test/index.tsx#L40))
- for `"should handle legacy string values"` test, cast string values `as unknown as number` to avoid changing `FocalPoint` type to _officially_ support `string` values as well as `number`

If the preference is to _officially_ support string values for focal points let me know and I'll update `FocalPoint` type to `Record< FocalPointAxis, number | string >` - however this will require further changes to the `FocalPointPicker` implementation.

## Testing Instructions

`npm run test:unit -- packages/components/src/focal-point-picker` tests pass (see screenshot below)
`npm run build:package-types` returns a zero exit code

## Screenshots or screencast 
<img width="844" alt="image" src="https://github.com/WordPress/gutenberg/assets/6638657/7b19dbee-c0f2-414d-9792-0ff69c76def6">
